docs/querying/api.md

```diff
@@ -336,6 +336,75 @@ $ curl http://localhost:9090/api/v1/label/job/values
 }
 ```
 
+## Querying exemplars
+The following endpoint returns a list of exemplars for a valid PromQL query during a specific time range:
+
+```
+GET /api/v1/query_exemplar
+POST /api/v1/query_exemplar
+```
+
+URL query parameters:
+
+- `query=<string>`: Prometheus expression query string.
+- `start=<rfc3339 | unix_timestamp>`: Start timestamp.
+- `end=<rfc3339 | unix_timestamp>`: End timestamp.
+
+Exemplar storage is not part of Prometheus' TSDB, so this endpoint doesn't allow filtering based on time.
+
+```json
+$ curl -g 'http://localhost:9090/api/v1/query_exemplar?query=test_exemplar_metric_total&start=2020-09-14T15:22:25.479Z&end=020-09-14T15:23:25.479Z'
+{
+    "status": "success",
+    "data": [
+        {
+            "seriesLabels": {
+                "__name__": "test_exemplar_metric_total",
+                "instance": "localhost:8090",
+                "job": "prometheus",
+                "service": "bar"
+            },
+            "exemplars": [
+                {
+                    "labels": {
+                        "traceID": "EpTxMJ40fUus7aGY"
+                    },
+                    "value": 6,
+                    "timestamp": 1600096945479,
+                    "hasTimestamp": true
```

Comment:
Right now I'm just returning a JSON encoded data from the `exemplar.Exemplar` struct. I don't actually understand why the field exists on that struct in the first place, not sure if it's part of the OM spec or if @shreyassrivatsan had another idea there? Maybe to indicate if the timestamp was exported by the source, instead of it being the scrape timestamp?

I can remove the field from the output for the API though.

Reply:
The exemplar timestamp is optional in the exposition. It is a different thing from the scrape/sample timestamp.

Reply:
Thanks for clarifying about the exposition.

Using the scrape/sample timestamp when an exported timestamp was not found allows users to query for exemplars during the same time range as their existing PromQL query. Both timestamps could be stored as a separate fields though.

Reply:
Would you send on a duplicate for every scrape, or just once?

Reply:
Like you mentioned in the other thread regarding the duplicate error (or not error), it's likely the exported exemplar for a series might not change from one scrape to the next, so for each subsequent scrape we'd skip storing the exemplar again until we see a new one.

Reply:
this would likely be a part of the same json marshal in the previous comment thread?